src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,177 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * Encapsulates all the interactions between the Duke interface and the user.
+ */
+public class Ui {
+    public static String BORDER = "_________________________________________________";
+
+    /**
+     * Returns a string with the inputted message on the Duke interface.
+     *
+     * @param msg A String to be printed.
+     * @return The string with a border.
+     */
+    public static String getMessage(String msg) {
+        return BORDER + "\n" + msg + "\n" + BORDER;
+    }
+
+    /**
+     * Returns a string with the Duke startup welcome message.
+     *
+     * @return The Duke welcome message string with a border.
+     */
+    public static String getWelcomeMessage() {
+        String welcomeMsg = "Hello! I'm duke.\n" + "What can I do for you?";
+        return getMessage(welcomeMsg);
+    }
+
+    /**
+     * Returns a string containing the Duke exit message.
+     *
+     * @return The Duke exit String with a border.
+     */
+    public String getExitMessage() {
+        String exitMsg = "Bye. Hope to see you again soon!";
+        return getMessage(exitMsg);
+    }
+
+    /**
+     * Returns a string containing the list of tasks stored in Duke.
+     *
+     * @param tasks A list of tasks stored.
+     * @return A String detailing the list of tasks stored with a border.
+     */
+    public String listTasks(ArrayList<Task> tasks) {
+        String reply = "Here are the tasks in your list:";
+        return getMessage(enumerateTasks(tasks, reply));
+    }
+
+    /**
+     * Returns a string containing the list of tasks that matches a certain requirement in Duke.
+     *
+     * @param tasks A list of tasks stored that matches a certain requirement.
+     * @return A String detailing the list of tasks that matches a certain requirement with a border.
+     */
+    public String listFoundTasks(ArrayList<Task> tasks) {
+        String reply = "Here are the matching tasks in your list:";
+        return getMessage(enumerateTasks(tasks, reply));
+    }
+
+    private String enumerateTasks(ArrayList<Task> tasks, String str) {
```

Comment:
I like how you separated the logic for enumerating tasks into its own method to improve code reuse!